db/statistics.go

```diff
@@ -508,7 +508,7 @@ func WriteValidatorBlockStats(day uint64) error {
 		(
 			select proposer, $3, sum(case when status = '1' then 1 else 0 end), sum(case when status = '2' then 1 else 0 end), sum(case when status = '3' then 1 else 0 end)
 			from blocks
-			where epoch >= $1 and epoch <= $2
+			where epoch >= $1 and epoch <= $2 and proposer != 2147483647
```

Comment:
Using the `/validator/{indexOrPubkey}/proposals` endpoint with `proposer` set to `2147483647` and epoch to something <= 100 will return data related to slot 0 and the proposer here is set to `2147483647`.

Instead of adding a `and proposer < 2147483647` to the query, you could maybe change `func parseApiValidatorParamToIndices` (and maybe also `func getAddressesOrIndicesFromAddressIndexOrPubkey`?) so that it does not add `2147483647` to the returned slice of indices. This would make sure that more API endpoints are less error prone.